build_tools/cmake/iree_cross_compile.cmake

```diff
@@ -0,0 +1,206 @@
+  iree_to_bool(config_enable_runtime_tracing "${IREE_${config_name}_ENABLE_RUNTIME_TRACING}")
+  iree_to_bool(config_enable_llvm "${IREE_${config_name}_ENABLE_LLVM}")
+  iree_to_bool(config_enable_emitc "${IREE_${config_name}_ENABLE_EMITC}")
+
+  iree_to_bool(config_build_compiler "${IREE_${config_name}_BUILD_COMPILER}")
+  iree_to_bool(config_build_tests "${IREE_${config_name}_BUILD_TESTS}")
+  iree_to_bool(config_build_docs "${IREE_${config_name}_BUILD_DOCS}")
+  iree_to_bool(config_build_samples "${IREE_${config_name}_BUILD_SAMPLES}")
+  iree_to_bool(config_build_debugger "${IREE_${config_name}_BUILD_DEBUGGER}")
+  iree_to_bool(config_build_python_bindings "${IREE_${config_name}_BUILD_PYTHON_BINDINGS}")
+  iree_to_bool(config_build_experimental "${IREE_${config_name}_BUILD_EXPERIMENTAL}")
+
+  # Escape semicolons in the targets list so that CMake doesn't expand them to
+  # spaces.
+  string(REPLACE ";" "$<SEMICOLON>" config_hal_drivers_to_build "${IREE_HAL_DRIVERS_TO_BUILD}")
+  string(REPLACE ";" "$<SEMICOLON>" config_compiler_targets_to_build "${IREE_TARGET_BACKENDS_TO_BUILD}")
+  # LINT.ThenChange(https://github.com/google/iree/tree/master/CMakeLists.txt:iree_options)
+
+  message(STATUS "C compiler for ${config_name} build: ${config_c_compiler}")
+  message(STATUS "C++ compiler for ${config_name} build: ${config_cxx_compiler}")
+
+  add_custom_command(OUTPUT ${IREE_${config_name}_BINARY_ROOT}/CMakeCache.txt
+    COMMAND "${CMAKE_COMMAND}" "${PROJECT_SOURCE_DIR}" -G "${CMAKE_GENERATOR}"
+        -DCMAKE_MAKE_PROGRAM="${CMAKE_MAKE_PROGRAM}"
+        -DCMAKE_BUILD_TYPE="${CMAKE_BUILD_TYPE}"
+        -DCMAKE_C_COMPILER="${config_c_compiler}"
+        -DCMAKE_CXX_COMPILER="${config_cxx_compiler}"
+        -DIREE_ENABLE_RUNTIME_TRACING=${config_enable_runtime_tracing}
+        -DIREE_ENABLE_LLVM=${config_enable_llvm}
+        -DIREE_ENABLE_EMITC=${config_enable_emitc}
+        -DIREE_BUILD_COMPILER=${config_build_compiler}
+        -DIREE_BUILD_TESTS=${config_build_tests}
+        -DIREE_BUILD_DOCS=${config_build_docs}
+        -DIREE_BUILD_SAMPLES=${config_build_samples}
+        -DIREE_BUILD_DEBUGGER=${config_build_debugger}
+        -DIREE_BUILD_PYTHON_BINDINGS=${config_build_python_bindings}
+        -DIREE_BUILD_EXPERIMENTAL=${config_build_experimental}
+        -DIREE_HAL_DRIVERS_TO_BUILD="${config_hal_drivers_to_build}"
+        -DIREE_TARGET_BACKENDS_TO_BUILD="${config_compiler_targets_to_build}"
+    WORKING_DIRECTORY ${config_binary_root}
+    DEPENDS iree_prepare_${config_name}_dir
+    COMMENT "Configuring IREE for ${config_name} build...")
+
+  add_custom_target(iree_configure_${config_name} DEPENDS ${config_binary_root}/CMakeCache.txt)
+endfunction()
+
+# iree_get_build_command
+#
+# Gets the CMake build command for the given `target`.
+#
+# Parameters:
+# - target: the target to build.
+# - bin_dir: root binary directory containing CMakeCache.txt.
+# - cmd_var: variable name for receiving the build command.
```

Comment:
(throughout this file)
How about using the same style as in IREE's other CMake parameter lists?

For example:
https://github.com/google/iree/blob/ccbe1c883f7672606861c7544dab133c8a6b4c14/build_tools/cmake/iree_cc_library.cmake#L21-L25

You could also use the same `cmake_parse_arguments` style: https://github.com/google/iree/blob/ccbe1c883f7672606861c7544dab133c8a6b4c14/build_tools/cmake/iree_cc_library.cmake#L67-L74

Reply:
Generally referring to keeping consistent style across CMake files:

1) Comment parameters as `# PARAM_NAME: description` (no leading `- `)
2) Naming parameters in `ALL_CAPS`
3) Putting arguments to `cmake_parse_arguments` on separate lines, to differentiate clearly between `<options>`, `<one_value_keywords>`, `<multi_value_keywords>`, etc. (see https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html) 

Concretely: in this function you can move `target bin_dir cmd_var` into the `<one_value_keywords>`, change those to `TARGET`/`BIN_DIR`/`CMD_VAR`, and then get them in the function as `${_RULE_TARGET}` etc.

Reply:
Capitalized variable names. 

Actually for parameter comments I'd think it's cleaner to prefix each one with `-`.. But if we decide the convention is not to do that, I'm fine with being consistent. So changed there.

Regarding `cmake_parse_arguments`, I guess there is a judgement call here as when we want to put all stuff in `cmake_parse_arguments`. I can see in `iree_cc_binary` we are using `NAME` instead of `TARGET` and synthesize the real target name under the hood. So it makes sense to differentiate. Here I'm actually "referencing" the target under host configuration. It's a bit different. I don't really feel using `NAME` would be good here. For target, normally CMake functions do not put the `TARGET` inside `cmake_parse_arguments`. Also for a function call typically just with one argument (e.g., `iree_declare_host_executable(iree-tblgen)`), it's a bit awkward to explicitly prefix the only argument with `TARGET` or something. So I'll follow that here to only put `TARGET` in function declaration and put all others in `cmake_parse_arguments`.